src/components/CountryPage/CountryPage.jsx

```diff
@@ -18,7 +18,9 @@ import Spinner from "../Spinner";
 export default function CountryPage() {
   const { isDarkMode } = useContext(DarkModeContext);
   const params = useParams();
-  const { details, loading } = useFetchSpecificCountry(params.countryCode);
+  const { countries, loading } = useFetchData(
```

Comment:
jak zrobiłeś taki uniwersalny hook (a to jest bardzo dobry pomysł) to warto zmienić też zwracane właściwości na bardzien uniwersalne, typu 'data' zamiast 'countires' - bo w tym przypadku countires nie jest do końca spójne z tym co tam rzeczywiście siedzi